src/Kafka/ConsumerGroups/components/ConsumerGroupResetOffset.tsx

```diff
@@ -0,0 +1,251 @@
+import {
+  Alert,
+  Button,
+  Checkbox,
+  Form,
+  FormGroup,
+  Modal,
+  ModalVariant,
+  Stack,
+  StackItem,
+  TextInput,
+  Title,
+} from "@patternfly/react-core";
+import { FunctionComponent } from "react";
+import { useTranslation } from "react-i18next";
+import "../ConsumerGroup.css";
+import { Consumer, OffsetValue } from "../types";
+import {
+  TableComposable,
+  Tbody,
+  Td,
+  Th,
+  Thead,
+  Tr,
+} from "@patternfly/react-table";
+import { TopicDropdown } from "./TopicsDropdown";
+import { OffsetDropdown } from "./OffsetDropdown";
+
+export type ConsumerRow = Consumer & {
+  selected?: boolean;
+};
+
+export type ConsumerGroupResetOffsetProps = {
+  isModalOpen: boolean;
+  isDisconnected: boolean;
+  groupId: string;
+  topics: string[];
+  selectedTopic: string;
+  customOffsetValue: string;
+  setcustomOffsetValue: (customOffsetValue: string) => void;
+  consumers: ConsumerRow[];
+  isSelected: boolean;
+  onClickClose: () => void;
+  onClickResetOffset: () => void;
+  onChangeTopic: (value: string) => void;
+  selectedOffset: OffsetValue;
+  onChangeOffsetValue: (value: OffsetValue) => void;
+  confirmCheckboxChecked: boolean;
+  onConfirmationChange: (value: boolean) => void;
+  SelectAllConsumer: (consumer: ConsumerRow[]) => void;
+  onSelectAll: (isSelected: boolean) => void;
+  onSelectRow: (index: number, isSelected: boolean) => void;
+};
+
+export const ConsumerGroupResetOffset: FunctionComponent<
+  ConsumerGroupResetOffsetProps
+> = ({
+  isDisconnected,
+  groupId,
+  topics,
+  customOffsetValue,
+  setcustomOffsetValue,
+  consumers,
+  onClickClose,
+  isModalOpen,
+  onClickResetOffset,
+  selectedTopic,
+  onChangeTopic,
+  selectedOffset,
+  onChangeOffsetValue,
+  confirmCheckboxChecked,
+  onConfirmationChange,
+  isSelected,
+  onSelectAll,
+  onSelectRow,
+}) => {
+  const { t } = useTranslation(["kafka"]);
+
+  const isResetOffsetDisabled = (): boolean => {
+    return (
+      !selectedTopic ||
+      !selectedOffset ||
+      !confirmCheckboxChecked ||
+      !isDisconnected ||
+      consumers.filter(({ selected }) => selected === true).length === 0
+    );
+  };
+
+  const tableColumns = {
+    partition: t("consumerGroup.partition"),
+    clientId: `${t("consumerGroup.client_id")} + ${t(
+      "consumerGroup.member_id"
+    )}`,
+    current_offset: t("consumerGroup.current_offset"),
+    log_end_offset: t("consumerGroup.log_end_offset"),
+    offset_lag: t("consumerGroup.offset_lag"),
+    new_offset: t("consumerGroup.new_offset"),
+  };
+
+  return (
+    <Modal
+      variant={ModalVariant.large}
+      isOpen={isModalOpen}
+      aria-label={"Modal for resetting offset of Kafka consumergroup"}
```

Comment:
Please, under translation.

Reply:
An `aria-label` isn't needed if the `title` prop is used on a modal. The PF Modal automatically uses the value provided for `title` as the accessible label for the modal.